File: dot-net-quiz/backend/Models/LaravelInterviewQuestion.cs
namespace backend.Models {
    public class LaravelInterviewQuestion {
        public int Id { get; set; }
        public string? Topic { get; set; }
        public string? Type { get; set; }
        public string? Question { get; set; }
        public string[]? Choices { get; set; }
        public int? CorrectAnswer { get; set; }
        public string? Explanation { get; set; }
    }
}

File: laravel-backend/app/Models/LaravelInterviewQuestion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;

class LaravelInterviewQuestion extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'topic',
        'type',
        'question',
        'choices',
        'correct_answer',
        'explanation',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'id' => 'integer',
        'choices' => 'array',
        'correct_answer' => 'integer',
    ];
}